tests/tests.js

```diff
@@ -257,7 +257,8 @@ var runTests = function runTests(arrayFrom) {
 		});
 
 		var hasMap = 'Map' in global;
-		t.test('works with Map objects', { 'skip': !hasMap }, function (te) {
+		var hasMapIterator = hasMap && ((hasSymbols && Map.prototype[Symbol.iterator]) || Map.prototype['@@iterator'] || Map.prototype.iterator);
```

Comment:
That's a good start - I believe there are also browsers that ship with `Map` but not `for..of`, so we need to be able to iterate those as well.